docs/index.rst

```diff
@@ -3,185 +3,288 @@ The confluent_kafka API
 
 A reliable, performant and feature rich Python client for Apache Kafka v0.8 and above.
 
-Clients
+Client API
+   - :ref:`AdminClient <pythonclient_adminclient>`
    - :ref:`Consumer <pythonclient_consumer>`
+   - :ref:`DeserializingConsumer <serde_consumer>`
    - :ref:`Producer <pythonclient_producer>`
-   - :ref:`AdminClient <pythonclient_adminclient>`
+   - :ref:`SerializingConsumer <serde_producer>`
+   - :ref:`SchemaRegistry <schemaregistry_client>`
+
+   - :ref:`AvroConsumer(Legacy) <avro_consumer>`
+   - :ref:`AvroProducer(Legacy) <avro_producer>`
+
+Serialization API
+    - :ref:`Deserializer <serde_deserializer>`
+    - :ref:`AvroDeserializer <schemaregistry_avro_deserializer>`
+    - :ref:`JSONDeserializer <schemaregistry_json_deserializer>`
+    - :ref:`ProtobufDeserializer <schemaregistry_protobuf_deserializer>`
+
+    - :ref:`Serializer <serde_Serializer>`
+    - :ref:`AvroSerializer <schemaregistry_avro_serializer>`
+    - :ref:`JSONSerializer <schemaregistry_json_serializer>`
+    - :ref:`ProtobufSerializer <schemaregistry_protobuf_serializer>`
 
+:ref:`Transactional Producer API <pythonclient_transactional>`
+
+:ref:`Kafka Client Configuration <pythonclient_configuration>`
 
 Supporting classes
     - :ref:`Message <pythonclient_message>`
     - :ref:`TopicPartition <pythonclient_topicpartition>`
+    - :ref:`SerializationContext <serde_ctx>`
+    - :ref:`Schema <schemaregistry_schema>`
+    - :ref:`RegisteredSchema <schemaregistry_registered_schema>`
+    - :ref:`SchemaRegistryError <schemaregistry_error>`
     - :ref:`KafkaError <pythonclient_kafkaerror>`
     - :ref:`KafkaException <pythonclient_kafkaexception>`
+    - :ref:`ConsumeError <pyclient_error_consumer>`
```

Comment:
I would group these a bit so that the SR stuff comes last, not in the middle, since everything else relates to the Kafka clients.

Reply:
ConsumeError is a KafkaClient error not a schema registry error.